app/kumactl/cmd/install/render_helm_files.go

```diff
@@ -0,0 +1,167 @@
+	}
+
+	overrideValues := generateOverrideValues(args)
+	if err := chartutil.ProcessDependencies(loadedChart, overrideValues); err != nil {
+		return nil, errors.Errorf("Failed to process dependencies: %s", err)
+	}
+
+	namespace := overrideValues["Namespace"].(string)
+	options := generateReleaseOptions(loadedChart.Metadata.Name, namespace)
+
+	valuesToRender, err := chartutil.ToRenderValues(loadedChart, overrideValues, options, nil)
+	if err != nil {
+		return nil, errors.Errorf("Failed to render values: %s", err)
+	}
+
+	files, err := engine.Render(loadedChart, valuesToRender)
+	if err != nil {
+		return nil, errors.Errorf("Failed to render templates: %s", err)
+	}
+	files["namespace.yaml"] = kumaSystemNamespace(namespace)
+
+	return postRender(loadedChart, files), nil
+}
+
+func loadCharts(templates []data.File) (*chart.Chart, error) {
+	files := []*loader.BufferedFile{}
+	for _, f := range templates {
+		files = append(files, &loader.BufferedFile{
+			Name: f.FullPath[1:],
+			Data: f.Data,
+		})
+	}
+
+	loadedChart, err := loader.LoadFiles(files)
+	if err != nil {
+		return nil, err
+	}
+
+	// Filter out the pre- templates
+	loadedTemplates := loadedChart.Templates
+	loadedChart.Templates = []*chart.File{}
+
+	for _, t := range loadedTemplates {
+		if !strings.HasPrefix(t.Name, "templates/pre-") {
+			loadedChart.Templates = append(loadedChart.Templates, &chart.File{
+				Name: t.Name,
+				Data: t.Data,
+			})
+		}
+	}
+
+	return loadedChart, nil
+}
+
+func generateOverrideValues(args interface{}) map[string]interface{} {
+	overrideValues := map[string]interface{}{}
+
+	v := reflect.ValueOf(args)
+	t := v.Type()
+	for i := 0; i < t.NumField(); i++ {
+		name := t.Field(i).Name
+		value := v.FieldByName(name).Interface()
+
+		splitName := strings.Split(name, "_")
+		len := len(splitName)
+
+		root := overrideValues
+
+		for i := 1; i < len-1; i++ {
+			n := splitName[i]
+
+			if _, ok := root[n]; !ok {
+				root[n] = map[string]interface{}{}
+			}
+			root = root[n].(map[string]interface{})
+		}
+		root[splitName[len-1]] = value
+	}
+
+	return overrideValues
+}
+
+func generateReleaseOptions(name, namespace string) chartutil.ReleaseOptions {
+	return chartutil.ReleaseOptions{
+		Name:      name,
+		Namespace: namespace,
+		Revision:  1,
+		IsInstall: true,
+		IsUpgrade: false,
+	}
+}
+
+func postRender(loadedChart *chart.Chart, files map[string]string) []data.File {
+	result := []data.File{}
+
+	for _, crd := range loadedChart.CRDObjects() {
+		result = append(result, data.File{
+			Data: crd.File.Data,
+			Name: crd.Name,
+		})
+	}
+
+	// sorted map of files to ensure consistency of the output
+	keys := make([]string, 0, len(files))
+	for k := range files {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	for _, k := range keys {
+		if strings.HasSuffix(k, "yaml") {
+			content := files[k]
+
+			// strip Helm Chart specific labels
+			for _, label := range stripLabels {
+				re := regexp.MustCompile("(?m)[\r\n]+^.*" + label + ".*$")
```

Comment:
nit - might want to compile these all ahead of time instead of for each file.

```go
func labelRegex(label string) *regexp.Regexp {
	return regexp.MustCompile("(?m)[\r\n]+^.*" + label + ".*$")
}

var stripLabelsRegexps = []*regexp.Regexp{
	labelRegex("app.kubernetes.io/managed-by"),
	labelRegex("helm.sh/chart"),
	labelRegex("app.kubernetes.io/version"),
}
```